modules/logging/ui/src/sap/commerce/toolset/logging/actionSystem/CxCreateCustomTemplateStateAction.kt

```diff
@@ -0,0 +1,174 @@
+/*
+ * This file is part of "SAP Commerce Developers Toolset" plugin for IntelliJ IDEA.
+ * Copyright (C) 2019-2025 EPAM Systems <hybrisideaplugin@epam.com> and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package sap.commerce.toolset.logging.actionSystem
+
+import com.intellij.openapi.actionSystem.ActionPlaces
+import com.intellij.openapi.actionSystem.ActionUpdateThread
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.Messages
+import com.intellij.ui.AnimatedIcon
+import com.intellij.util.asSafely
+import sap.commerce.toolset.HybrisIcons
+import sap.commerce.toolset.logging.CxRemoteLogStateService
+import sap.commerce.toolset.logging.custom.CxCustomLogTemplateService
+import sap.commerce.toolset.logging.custom.settings.CxCustomLogTemplatesSettings
+import sap.commerce.toolset.logging.presentation.CxLoggerPresentation
+import sap.commerce.toolset.logging.selectedNodes
+import sap.commerce.toolset.logging.ui.CxAdvancedCustomLogTemplateDialog
+import sap.commerce.toolset.logging.ui.CxCustomLogTemplateDialog
+import sap.commerce.toolset.logging.ui.LogTemplateDialogContext
+import sap.commerce.toolset.logging.ui.tree.nodes.CxBundledLogTemplateItemNode
+import sap.commerce.toolset.logging.ui.tree.nodes.CxCustomLogTemplateItemNode
+import sap.commerce.toolset.logging.ui.tree.nodes.CxRemoteLogStateNode
+
+class CxCreateCustomTemplateStateAction : AnAction() {
+
+    override fun getActionUpdateThread() = ActionUpdateThread.BGT
+
+    override fun actionPerformed(e: AnActionEvent) {
+        e.presentation.isVisible = ActionPlaces.ACTION_SEARCH != e.place
+        if (!e.presentation.isVisible) return
+
+        val project = e.project ?: return
+
+        val customLogTemplateService = CxCustomLogTemplateService.getInstance(project)
+
+        val selectedNodes = e.selectedNodes() ?: return
+        if (selectedNodes.groupBy { it.javaClass }.keys.size > 1) {
+            Messages.showErrorDialog("A custom template cannot be created from the current selection.", "Incompatible Nodes Selected")
+            return
+        }
+
+        val dialogTitle = getDialogTitle(selectedNodes)
+        val templateName = templateName(project, selectedNodes)
```

Comment:
what will happen if we select multiple nodes of different types for those methods dependent on `selectedNodes.first()`?